<a href="https://colab.research.google.com/github/smata2201/Progra_2_practica/blob/main/Proyecto_1_SantiagoMata_AkillKenayChacon.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Proyecto 1
Santiago Mata Alvarado C4G964

Akill Kenay Chacón Chacón C5E118

## Desigualdades educativas y socioeconómicas entre las zonas urbana y rural de Costa Rica durante 2021–2022

El proyecto busca analizar si existe un efecto notable en los sectores académicos, laborales y socioeconómicos según la zona a la que una persona pertenezca, analizando así bases de datos del INEC y el MEP para poder así buscar incongruencias entre estas dos zonas (urbana y rural).

## Preguntas guía

¿Qué diferencias se observan en el rendimiento académico entre las zonas urbana y rural durante 2021–2022?

¿Cómo variaron los indicadores de empleo y pobreza entre las zonas urbana y rural durante 2021–2022?

¿Qué relación se observa entre el rendimiento académico y las condiciones socioeconómicas de las zonas urbana y rural al integrar los datos del MEP y del INEC?

In [113]:
import pandas as pd
import numpy as np

## Datos de rendimiento académico del MEP

Primero se realiza la lectura y preparación de los datos del MEP correspondientes al rendimiento académico. Se seleccionan los años 2021 y 2022 y se comparan los resultados entre las zonas urbana y rural.

In [114]:
mep = pd.read_excel("/content/AprobadosAplazadosReprobados14-22.xlsx")

# realizo una primera lectura para observar como viene organizado el archivo

In [115]:
mep.head(10)

,"Matrícula final y rendimiento, III Ciclo y Educación Diversificada, 2014-2022 (rendimiento antes de convocatorias)",Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,...,Unnamed: 65,Unnamed: 66,Unnamed: 67,Unnamed: 68,Unnamed: 69,Unnamed: 70,Unnamed: 71,Unnamed: 72,Unnamed: 73,Unnamed: 74
0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,8°,NaN,9°,NaN,10°,NaN,11°,NaN,12°,NaN
1,CURSO LECTIVO,CODIGO,NOMBRE,REGION,CIRES,PR,CAN,DIS,PROVINCIA,CANTON,...,RE2T,RE2H,RE3T,RE3H,RE4T,RE4H,RE5T,RE5H,RE6T,RE6H
2,2014,5072,COLEGIO DE GRAVILIAS,DESAMPARADOS,01,1,03,12,SAN JOSE,DESAMPARADOS,...,43,24,16,11,47,26,11,5,0,0
3,2014,3984,COLEGIO NUESTRA SEÑORA,DESAMPARADOS,01,1,03,01,SAN JOSE,DESAMPARADOS,...,7,0,3,0,9,0,0,0,0,0
4,2014,3988,LICEO SAN ANTONIO,DESAMPARADOS,01,1,03,05,SAN JOSE,DESAMPARADOS,...,25,14,24,11,50,34,26,16,0,0
5,2014,6108,I.E.G.B. REPÚBLICA DE PANAMÁ,DESAMPARADOS,01,1,03,05,SAN JOSE,DESAMPARADOS,...,15,13,3,3,0,0,0,0,0,0
6,2014,0000,SISTEMA EDUCATIVO BENDITO JESUS,DESAMPARADOS,01,1,03,05,SAN JOSE,DESAMPARADOS,...,0,0,0,0,0,0,0,0,0,0
7,2014,3985,LICEO SAN MIGUEL,DESAMPARADOS,02,1,03,02,SAN JOSE,DESAMPARADOS,...,42,13,21,10,8,3,23,5,0,0
8,2014,3995,LICEO JOAQUIN GUTIERREZ MANGEL,DESAMPARADOS,02,1,03,04,SAN JOSE,DESAMPARADOS,...,27,17,4,2,11,7,1,0,0,0
9,2014,5870,UNIDAD PEDAGÓGICA SOTERO GONZÁLEZ B,DESAMPARADOS,02,1,03,03,SAN JOSE,DESAMPARADOS,...,10,5,2,2,0,0,0,0,0,0


In [116]:
mep = pd.read_excel(
    "/content/AprobadosAplazadosReprobados14-22.xlsx",
    header=2
)

# utilizo la tercera fila del archivo como encabezado de las columnas

In [117]:
mep.head(10)

,CURSO LECTIVO,CODIGO,NOMBRE,REGION,CIRES,PR,CAN,DIS,PROVINCIA,CANTON,...,RE2T,RE2H,RE3T,RE3H,RE4T,RE4H,RE5T,RE5H,RE6T,RE6H
0,2014,5072,COLEGIO DE GRAVILIAS,DESAMPARADOS,1,1,3,12,SAN JOSE,DESAMPARADOS,...,43.0,24.0,16.0,11.0,47.0,26.0,11.0,5.0,0.0,0.0
1,2014,3984,COLEGIO NUESTRA SEÑORA,DESAMPARADOS,1,1,3,1,SAN JOSE,DESAMPARADOS,...,7.0,0.0,3.0,0.0,9.0,0.0,0.0,0.0,0.0,0.0
2,2014,3988,LICEO SAN ANTONIO,DESAMPARADOS,1,1,3,5,SAN JOSE,DESAMPARADOS,...,25.0,14.0,24.0,11.0,50.0,34.0,26.0,16.0,0.0,0.0
3,2014,6108,I.E.G.B. REPÚBLICA DE PANAMÁ,DESAMPARADOS,1,1,3,5,SAN JOSE,DESAMPARADOS,...,15.0,13.0,3.0,3.0,0.0,0.0,0.0,0.0,0.0,0.0
4,2014,0,SISTEMA EDUCATIVO BENDITO JESUS,DESAMPARADOS,1,1,3,5,SAN JOSE,DESAMPARADOS,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
5,2014,3985,LICEO SAN MIGUEL,DESAMPARADOS,2,1,3,2,SAN JOSE,DESAMPARADOS,...,42.0,13.0,21.0,10.0,8.0,3.0,23.0,5.0,0.0,0.0
6,2014,3995,LICEO JOAQUIN GUTIERREZ MANGEL,DESAMPARADOS,2,1,3,4,SAN JOSE,DESAMPARADOS,...,27.0,17.0,4.0,2.0,11.0,7.0,1.0,0.0,0.0,0.0
7,2014,5870,UNIDAD PEDAGÓGICA SOTERO GONZÁLEZ B,DESAMPARADOS,2,1,3,3,SAN JOSE,DESAMPARADOS,...,10.0,5.0,2.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0
8,2014,5819,UNIDAD PEDAGÓGICA JOSÉ MARÍA ZELEDÓ,DESAMPARADOS,2,1,3,2,SAN JOSE,DESAMPARADOS,...,0.0,0.0,3.0,3.0,0.0,0.0,0.0,0.0,0.0,0.0
9,2014,6027,LICEO HIGUITO,DESAMPARADOS,2,1,3,2,SAN JOSE,DESAMPARADOS,...,12.0,5.0,1.0,0.0,11.0,8.0,0.0,0.0,0.0,0.0


In [118]:
mep.columns

Index(['CURSO LECTIVO', 'CODIGO', 'NOMBRE', 'REGION', 'CIRES', 'PR', 'CAN',
       'DIS', 'PROVINCIA', 'CANTON', 'DISTRITO', 'POBLADO', 'SECTOR', 'ZONA',
       'RAMA', 'MFT', 'MFH', 'MFM', 'MF1T', 'MF1H', 'MF2T', 'MF2H', 'MF3T',
       'MF3H', 'MF4T', 'MF4H', 'MF5T', 'MF5H', 'MF6T', 'MF6H', 'APT', 'APH',
       'APM', 'AP1T', 'AP1H', 'AP2T', 'AP2H', 'AP3T', 'AP3H', 'AP4T', 'AP4H',
       'AP5T', 'AP5H', 'AP6T', 'AP6H', 'AZT', 'AZH', 'AZM', 'AZ1T', 'AZ1H',
       'AZ2T', 'AZ2H', 'AZ3T', 'AZ3H', 'AZ4T', 'AZ4H', 'AZ5T', 'AZ5H', 'AZ6T',
       'AZ6H', 'RET', 'REH', 'REM', 'RE1T', 'RE1H', 'RE2T', 'RE2H', 'RE3T',
       'RE3H', 'RE4T', 'RE4H', 'RE5T', 'RE5H', 'RE6T', 'RE6H'],
      dtype='object')

In [119]:
mep["CURSO LECTIVO"].value_counts()

,count
CURSO LECTIVO,
2020,984
2022,983
2021,973
2019,968
2017,965
2016,959
2018,957
2014,952
2015,949


In [120]:
mep["ZONA"].value_counts()

,count
ZONA,
1,5553
2,3137


In [121]:
mep_21_22 = mep[(mep["CURSO LECTIVO"] == 2021) |
                (mep["CURSO LECTIVO"] == 2022)].copy()

# selecciono solamente los datos de 2021 y 2022

In [122]:
mep_21_22["CURSO LECTIVO"].value_counts()

# comprobamos equivalencia en los datos

,count
CURSO LECTIVO,
2022,983
2021,973


In [123]:
mep_21_22.loc[mep_21_22["ZONA"] == 1, "ZONA"] = "Urbana"
mep_21_22.loc[mep_21_22["ZONA"] == 2, "ZONA"] = "Rural"

# cambio los valores de zona para trabajar con Urbana y Rural

/tmp/ipykernel_5571/266415434.py:1: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'Urbana' has dtype incompatible with int64, please explicitly cast to a compatible dtype first.
  mep_21_22.loc[mep_21_22["ZONA"] == 1, "ZONA"] = "Urbana"


In [124]:
mep_21_22["ZONA"].value_counts()

# comprobamos

,count
ZONA,
Urbana,1249
Rural,707


In [125]:
mep_21_22[
    ["CURSO LECTIVO", "ZONA", "MFT", "APT", "AZT", "RET"]
].head(10)

# selecciono las variables de rendimiento que voy a utilizar

,CURSO LECTIVO,ZONA,MFT,APT,AZT,RET
6734,2021,Urbana,492.0,198.0,294.0,0.0
6735,2021,Urbana,705.0,705.0,0.0,0.0
6736,2021,Urbana,999.0,611.0,388.0,0.0
6737,2021,Urbana,243.0,243.0,0.0,0.0
6738,2021,Urbana,811.0,485.0,326.0,0.0
6739,2021,Urbana,1285.0,1177.0,108.0,0.0
6740,2021,Urbana,290.0,262.0,28.0,0.0
6741,2021,Urbana,319.0,297.0,22.0,0.0
6742,2021,Urbana,1117.0,562.0,555.0,0.0
6743,2021,Urbana,843.0,354.0,489.0,0.0


In [126]:
rendimiento = mep_21_22.groupby(
    ["CURSO LECTIVO", "ZONA"]
)[["MFT", "APT", "AZT", "RET"]].sum()

rendimiento

# agrupo los datos por año y zona para comparar el rendimiento

MFT       APT      AZT    RET
CURSO LECTIVO ZONA                                      
2021          Rural    98618.0   74460.0  24158.0    0.0
              Urbana  301856.0  224383.0  77444.0   29.0
2022          Rural    96624.0   72595.0  24029.0    0.0
              Urbana  291151.0  214045.0  76983.0  123.0

In [127]:
rendimiento["PORCENTAJE_APROBADOS"] = (
    rendimiento["APT"] / rendimiento["MFT"]
) * 100

rendimiento

MFT       APT      AZT    RET  PORCENTAJE_APROBADOS
CURSO LECTIVO ZONA                                                            
2021          Rural    98618.0   74460.0  24158.0    0.0             75.503458
              Urbana  301856.0  224383.0  77444.0   29.0             74.334451
2022          Rural    96624.0   72595.0  24029.0    0.0             75.131437
              Urbana  291151.0  214045.0  76983.0  123.0             73.516835

In [128]:
rendimiento["PORCENTAJE_APLAZADOS"] = (
    rendimiento["AZT"] / rendimiento["MFT"]
) * 100

rendimiento

MFT       APT      AZT    RET  \
CURSO LECTIVO ZONA                                         
2021          Rural    98618.0   74460.0  24158.0    0.0   
              Urbana  301856.0  224383.0  77444.0   29.0   
2022          Rural    96624.0   72595.0  24029.0    0.0   
              Urbana  291151.0  214045.0  76983.0  123.0   

                      PORCENTAJE_APROBADOS  PORCENTAJE_APLAZADOS  
CURSO LECTIVO ZONA                                                
2021          Rural              75.503458             24.496542  
              Urbana             74.334451             25.655942  
2022          Rural              75.131437             24.868563  
              Urbana             73.516835             26.440919

In [129]:
rendimiento_final = rendimiento.reset_index()

rendimiento_final

,CURSO LECTIVO,ZONA,MFT,APT,AZT,RET,PORCENTAJE_APROBADOS,PORCENTAJE_APLAZADOS
0,2021,Rural,98618.0,74460.0,24158.0,0.0,75.503458,24.496542
1,2021,Urbana,301856.0,224383.0,77444.0,29.0,74.334451,25.655942
2,2022,Rural,96624.0,72595.0,24029.0,0.0,75.131437,24.868563
3,2022,Urbana,291151.0,214045.0,76983.0,123.0,73.516835,26.440919


In [130]:
rendimiento_final = rendimiento_final.rename(
    columns={
        "CURSO LECTIVO": "Año",
        "ZONA": "Zona"
    }
)

rendimiento_final

# cambio los nombres de las columnas para poder unir las bases posteriormente

,Año,Zona,MFT,APT,AZT,RET,PORCENTAJE_APROBADOS,PORCENTAJE_APLAZADOS
0,2021,Rural,98618.0,74460.0,24158.0,0.0,75.503458,24.496542
1,2021,Urbana,301856.0,224383.0,77444.0,29.0,74.334451,25.655942
2,2022,Rural,96624.0,72595.0,24029.0,0.0,75.131437,24.868563
3,2022,Urbana,291151.0,214045.0,76983.0,123.0,73.516835,26.440919


## Datos de empleo del INEC

Para analizar las condiciones laborales de las zonas urbana y rural se utilizan
los datos de la Encuesta Continua de Empleo del INEC para los años 2021 y 2022.

In [131]:
empleo = pd.read_excel("/content/seempleoiiitri2010-mjj2026-01.xlsx")

# realizo una primera lectura para observar como viene organizado el archivo

In [132]:
pd.ExcelFile("/content/seempleoiiitri2010-mjj2026-01.xlsx").sheet_names

#me interesa separar en urbano y rural para el proyecto

['Índice sitio Web',
 'C1 total ',
 'C1 urbano',
 'C1 rural',
 'C1 hombre',
 'C1 mujer']

In [133]:
empleo_urbano = pd.read_excel(
    "/content/seempleoiiitri2010-mjj2026-01.xlsx",
    sheet_name="C1 urbano"
)

empleo_urbano.head(15)

#las primeras lineas de documento no me aportan informacion por ser encabezados

,Cuadro 1.1,Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,...,Unnamed: 182,Unnamed: 183,Unnamed: 184,Unnamed: 185,Unnamed: 186,Unnamed: 187,Unnamed: 188,Unnamed: 189,Unnamed: 190,Unnamed: 191
0,Costa Rica zona urbana: Población nacional seg...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Indicadores generales,Trimestre,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,NaN,JAS 2010,ASO 2010,SON 2010,OND 2010,NDE 2010,DEF 2010,EFM 2011,FMA 2011,MAM 2011,...,ASO 2025,SON 2025,OND 2025,NDE 2025,DEF 2025,EFM 2026,FMA 2026,MAM 2026,AMJ 2026,MJJ 2026
4,NaN,Jul - Sep,Ago - Oct,Sep - Nov,Oct - Dic,Nov - Ene,Dic - Feb,Ene - Mar,Feb - Abr,Mar - May,...,Ago - Oct,Set - Nov,Oct - Dic,Nov - Ene,Dic - Feb,Ene - Mar,Feb - Abr,Mar - May,Abr - Jun,May - Jul
5,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,1. Población total urbana,3304084,3306716,3310788,3314391,3317821,3320800,3324247,3326947,3331672,...,3883043,3885154,3887922,3890512,3893434,3895733,3898357,3900930,3903364,3905457
7,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8,1.1 Grupo de edad,3304084,3306716,3310788,3314391,3317821,3320800,3324247,3326947,3331672,...,3883043,3885154,3887922,3890512,3893434,3895733,3898357,3900930,3903364,3905457
9,De 0 a 14 años,793105.0,792252.0,791628.0,790806.0,789254.0,788253,787677,786898.0,786322.0,...,755562,755456,755210,755064,755292,755229,754979,754422,754414,754049


In [134]:
empleo_urbano = pd.read_excel(
    "/content/seempleoiiitri2010-mjj2026-01.xlsx",
    sheet_name="C1 urbano"
)

# leo los datos correspondientes a la zona urbana

In [135]:
empleo_rural = pd.read_excel(
    "/content/seempleoiiitri2010-mjj2026-01.xlsx",
    sheet_name="C1 rural"
)

empleo_rural.head(15)

,Cuadro 1.2,Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,...,Unnamed: 182,Unnamed: 183,Unnamed: 184,Unnamed: 185,Unnamed: 186,Unnamed: 187,Unnamed: 188,Unnamed: 189,Unnamed: 190,Unnamed: 191
0,Costa Rica zona rural: Población nacional segú...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Indicadores generales,Trimestre,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,NaN,JAS 2010,ASO 2010,SON 2010,OND 2010,NDE 2010,DEF 2010,EFM 2011,FMA 2011,MAM 2011,...,ASO 2025,SON 2025,OND 2025,NDE 2025,DEF 2025,EFM 2026,FMA 2026,MAM 2026,AMJ 2026,MJJ 2026
4,NaN,Jul - Sep,Ago - Oct,Sep - Nov,Oct - Dic,Nov - Ene,Dic - Feb,Ene - Mar,Feb - Abr,Mar - May,...,Ago - Oct,Set - Nov,Oct - Dic,Nov - Ene,Dic - Feb,Ene - Mar,Feb - Abr,Mar - May,Abr - Jun,May - Jul
5,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,1. Población total rural,1235351,1236052,1237958,1239522,1240422,1242400,1243217,1245402,1245952,...,1479891,1481253,1482211,1483022,1484979,1485585,1487019,1488441,1489166,1490796
7,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8,1.1 Grupo de edad,1235351,1236052,1237958,1239522,1240422,1242400,1243217,1245402,1245952,...,1479891,1481253,1482211,1483022,1484979,1485585,1487019,1488441,1489166,1490796
9,De 0 a 14 años,348150.0,348099,347915,347600.0,347311.0,347123.0,346685.0,346424.0,346419.0,...,339828,339865,339754,339715,339793,339666,339784,339799,339649,339712


In [136]:
empleo_urbano.iloc[:, 0].dropna()

,Cuadro 1.1
0,Costa Rica zona urbana: Población nacional seg...
2,Indicadores generales
6,1. Población total urbana
8,1.1 Grupo de edad
9,De 0 a 14 años
...,...
130,13/ Porcentaje que representa la población ocu...
131,14/ Porcentaje de la población ocupada que rec...
132,15/ Porcentaje de la población desempleada más...
133,* Coeficiente de variación mayor al 20%. El co...


In [137]:
empleo_rural.iloc[:, 0].dropna()

,Cuadro 1.2
0,Costa Rica zona rural: Población nacional segú...
2,Indicadores generales
6,1. Población total rural
8,1.1 Grupo de edad
9,De 0 a 14 años
...,...
130,13/ Porcentaje que representa la población ocu...
131,14/ Porcentaje de la población ocupada que rec...
132,15/ Porcentaje de la población desempleada más...
133,* Coeficiente de variación mayor al 20%. El co...


In [138]:
for i in range(len(empleo_urbano.columns)):
    trimestre = empleo_urbano.iloc[3, i]

    if "2021" in str(trimestre) or "2022" in str(trimestre):
        print(i, trimestre)
        #identifico las columnas en trimestre que abarcan 2021 a 2022

127 EFM 2021
128 FMA 2021
129 MAM 2021
130 AMJ 2021
131 MJJ 2021
132 JJA 2021
133 JAS 2021
134 ASO 2021
135 SON 2021
136 OND 2021
137 NDE 2021
138 DEF 2021
139 EFM 2022
140 FMA 2022
141 MAM 2022
142 AMJ 2022
143 MJJ 2022
144 JJA 2022
145 JAS 2022
146 ASO 2022
147 SON 2022
148 OND 2022
149 NDE 2022
150 DEF 2022


In [139]:
for i in range(len(empleo_rural.columns)):
    trimestre = empleo_rural.iloc[3, i]

    if "2021" in str(trimestre) or "2022" in str(trimestre):
        print(i, trimestre)
        #identifico de forma paralela en rural

127 EFM 2021
128 FMA 2021
129 MAM 2021
130 AMJ 2021
131 MJJ 2021
132 JJA 2021
133 JAS 2021
134 ASO 2021
135 SON 2021
136 OND 2021
137 NDE 2021
138 DEF 2021
139 EFM 2022
140 FMA 2022
141 MAM 2022
142 AMJ 2022
143 MJJ 2022
144 JJA 2022
145 JAS 2022
146 ASO 2022
147 SON 2022
148 OND 2022
149 NDE 2022
150 DEF 2022


In [140]:
print(empleo_urbano.iloc[[104, 105, 109], 127:151])
print(empleo_rural.iloc[[104, 105, 109], 127:151])
#aqui separo mis indicadores de preferencia, en las columnas que contienen 2021 y 2022 que sacamos antes

    Unnamed: 127 Unnamed: 128 Unnamed: 129 Unnamed: 130 Unnamed: 131  \
104    61.884428    61.576926     61.54213    60.918907    61.343438   
105    49.574184    50.438865    50.257818    49.399832    50.285913   
109    19.892312    18.088042    18.335914    18.908867    18.025603   

    Unnamed: 132 Unnamed: 133 Unnamed: 134 Unnamed: 135 Unnamed: 136  ...  \
104    62.047593    62.712268    61.957148    61.301645    60.989696  ...   
105    51.639569     53.01163      52.5071    52.185578    52.455751  ...   
109     16.77426    15.468486    15.252555    14.870836    13.992438  ...   

    Unnamed: 141 Unnamed: 142 Unnamed: 143 Unnamed: 144 Unnamed: 145  \
104    61.025553    60.475152    61.339286    61.421992    61.756141   
105    53.744313    53.354986     53.93541    54.108705    54.462828   
109    11.931461    11.773705    12.070365    11.906626    11.809859   

    Unnamed: 146 Unnamed: 147 Unnamed: 148 Unnamed: 149 Unnamed: 150  
104      61.6212    61.122899    60.797184

In [141]:
prom_urbano_2021 = empleo_urbano.iloc[[104, 105, 109], 127:139].mean(axis=1)
prom_urbano_2022 = empleo_urbano.iloc[[104, 105, 109], 139:151].mean(axis=1)
#promedio las tasas que seleccione para no tener 12 trimestres por periodo
print(prom_urbano_2021)
print(prom_urbano_2022)
prom_rural_2021 = empleo_rural.iloc[[104, 105, 109], 127:139].mean(axis=1)
prom_rural_2022 = empleo_rural.iloc[[104, 105, 109], 139:151].mean(axis=1)

print(prom_rural_2021)
print(prom_rural_2022)

104    61.436052
105    51.380254
109     16.35962
dtype: object
104    60.899214
105    53.510157
109     12.13331
dtype: object
104    56.595429
105    48.467834
109    14.367144
dtype: object
104    56.698778
105    50.195066
109    11.466442
dtype: object


In [142]:
resumen_empleo = pd.DataFrame({
    "Año": [2021, 2021, 2022, 2022],
    "Zona": ["Urbana", "Rural", "Urbana", "Rural"],
    "Tasa_participacion": [
        prom_urbano_2021.iloc[0],
        prom_rural_2021.iloc[0],
        prom_urbano_2022.iloc[0],
        prom_rural_2022.iloc[0]
    ],
    "Tasa_ocupacion": [
        prom_urbano_2021.iloc[1],
        prom_rural_2021.iloc[1],
        prom_urbano_2022.iloc[1],
        prom_rural_2022.iloc[1]
    ],
    "Tasa_desempleo": [
        prom_urbano_2021.iloc[2],
        prom_rural_2021.iloc[2],
        prom_urbano_2022.iloc[2],
        prom_rural_2022.iloc[2]
    ]
})

resumen_empleo

,Año,Zona,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo
0,2021,Urbana,61.436052,51.380254,16.359620
1,2021,Rural,56.595429,48.467834,14.367144
2,2022,Urbana,60.899214,53.510157,12.133310
3,2022,Rural,56.698778,50.195066,11.466442


In [143]:
#transformo la tabla a un CSV
resumen_empleo.to_csv(
    "/content/resumen_empleo.csv",
    index=False,
    encoding="utf-8-sig"
)

empleo_csv = pd.read_csv("/content/resumen_empleo.csv")

empleo_csv

,Año,Zona,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo
0,2021,Urbana,61.436052,51.380254,16.359620
1,2021,Rural,56.595429,48.467834,14.367144
2,2022,Urbana,60.899214,53.510157,12.133310
3,2022,Rural,56.698778,50.195066,11.466442


## Datos de pobreza del INEC

En esta sección se utilizan los datos de la Encuesta Nacional de Hogares (ENAHO) del INEC para analizar los niveles de pobreza en las zonas urbana y rural durante los años 2021 y 2022.

In [144]:
pobreza = pd.read_excel(
    "/content/reenaho2022-linea_de_pobreza.xlsx",
    sheet_name="6"
)
# utilizo la hoja 6 porque contiene los datos de pobreza separados por año y zona

/usr/local/lib/python3.13/dist-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Unknown extension is not supported and will be removed
  warn(msg)
/usr/local/lib/python3.13/dist-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Conditional Formatting extension is not supported and will be removed
  warn(msg)


In [145]:
pobreza.head()

,CUADRO 6,Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5
0,Costa Rica. Distribución relativa de los hogar...,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN
2,Zona y año,Total,No pobres,Pobres,NaN,NaN
3,NaN,NaN,NaN,Total,Pobreza no extrema,Pobreza extrema
4,NaN,NaN,NaN,NaN,NaN,NaN


In [146]:
pobreza.iloc[:, 0].dropna()
# reviso la primera columna para identificar donde estan los datos deseados de zona y año

,CUADRO 6
0,Costa Rica. Distribución relativa de los hogar...
2,Zona y año
6,Total
8,2010
9,2011
10,2012
11,2013
12,2014
13,2015
14,2016


In [147]:
pobreza_21_22 = pobreza.iloc[[35, 36, 51, 52], :].copy()
# selecciono los datos de 2021 y 2022 para zona urbana y rural

In [148]:
pobreza_21_22 = pobreza_21_22.reset_index(drop=True)
# reinicio los indices de la tabla

In [149]:
pobreza_21_22.columns = [
    "Año",
    "Total",
    "No_pobres",
    "Pobres",
    "Pobreza_no_extrema",
    "Pobreza_extrema"
]
# cambio los nombres de las columnas para identificar mejor las variables

In [150]:
pobreza_21_22["Zona"] = ["Urbana", "Urbana", "Rural", "Rural"]
# agrego la zona correspondiente a cada año

In [151]:
pobreza_21_22 = pobreza_21_22.drop(columns=["Total"])
# elimino la columna total porque todos sus valores corresponden a 100

In [152]:
pobreza_21_22 = pobreza_21_22.reset_index(drop=True)
# reinicio los indices de la tabla

In [153]:
# convierto los datos a json para cumplir con el requisito de utilizar
# distintos formatos, ya que las fuentes encontradas estaban en excel

pobreza_21_22.to_json(
    "/content/pobreza_21_22.json",
    orient="records",
    force_ascii=False
)

In [154]:
with open("/content/pobreza_21_22.json", "r") as archivo:
    pobreza_json = pd.read_json(archivo)

# abro el archivo json y leo los datos de pobreza

In [155]:
pobreza_json

,Año,No_pobres,Pobres,Pobreza_no_extrema,Pobreza_extrema,Zona
0,2021,78.233883,21.766117,16.266310,5.499806,Urbana
1,2022,78.945941,21.054059,15.412405,5.641654,Urbana
2,2021,73.651675,26.348325,18.007161,8.341164,Rural
3,2022,71.736411,28.263589,19.935152,8.328437,Rural


In [156]:
pobreza_json[["Año", "Zona", "Pobres"]]

,Año,Zona,Pobres
0,2021,Urbana,21.766117
1,2022,Urbana,21.054059
2,2021,Rural,26.348325
3,2022,Rural,28.263589


In [157]:
comparacion_pobreza = pobreza_json.set_index(
    ["Año", "Zona"]
)["Pobres"].unstack()

comparacion_pobreza

Zona,Rural,Urbana
Año,,
2021,26.348325,21.766117
2022,28.263589,21.054059


In [158]:
tipos_pobreza = pobreza_json.set_index(
    ["Año", "Zona"]
)[["Pobreza_no_extrema", "Pobreza_extrema"]]

tipos_pobreza

,,Pobreza_no_extrema,Pobreza_extrema
Año,Zona,,
2021,Urbana,16.266310,5.499806
2022,Urbana,15.412405,5.641654
2021,Rural,18.007161,8.341164
2022,Rural,19.935152,8.328437


## Integración de las bases de datos

En esta sección se integran los datos de rendimiento académico, empleo y pobreza utilizando como variables en común el año y la zona.

In [159]:
datos_finales = rendimiento_final.merge(
    empleo_csv,
    on=["Año", "Zona"],
    how="inner",
    validate="one_to_one",
    indicator=True
)
# uno los datos de rendimiento y empleo utilizando el año y la zona
# verifico que cada combinación aparezca una sola vez en ambas tablas

datos_finales

,Año,Zona,MFT,APT,AZT,RET,PORCENTAJE_APROBADOS,PORCENTAJE_APLAZADOS,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo,_merge
0,2021,Rural,98618.0,74460.0,24158.0,0.0,75.503458,24.496542,56.595429,48.467834,14.367144,both
1,2021,Urbana,301856.0,224383.0,77444.0,29.0,74.334451,25.655942,61.436052,51.380254,16.359620,both
2,2022,Rural,96624.0,72595.0,24029.0,0.0,75.131437,24.868563,56.698778,50.195066,11.466442,both
3,2022,Urbana,291151.0,214045.0,76983.0,123.0,73.516835,26.440919,60.899214,53.510157,12.133310,both


In [160]:
datos_finales["_merge"].value_counts()
# reviso que todos los datos hayan coincidido en ambas tablas

,count
_merge,
both,4
left_only,0
right_only,0


In [161]:
datos_finales = datos_finales.drop(columns=["_merge"])
# elimino la columna del indicador luego de comprobar que la union fue correcta

In [162]:
datos_finales = datos_finales.merge(
    pobreza_json,
    on=["Año", "Zona"],
    how="inner",
    validate="one_to_one",
    indicator=True
)
# uno los datos de rendimiento y empleo con los datos de pobreza
# utilizando nuevamente el año y la zona

datos_finales

,Año,Zona,MFT,APT,AZT,RET,PORCENTAJE_APROBADOS,PORCENTAJE_APLAZADOS,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo,No_pobres,Pobres,Pobreza_no_extrema,Pobreza_extrema,_merge
0,2021,Rural,98618.0,74460.0,24158.0,0.0,75.503458,24.496542,56.595429,48.467834,14.367144,73.651675,26.348325,18.007161,8.341164,both
1,2021,Urbana,301856.0,224383.0,77444.0,29.0,74.334451,25.655942,61.436052,51.380254,16.359620,78.233883,21.766117,16.266310,5.499806,both
2,2022,Rural,96624.0,72595.0,24029.0,0.0,75.131437,24.868563,56.698778,50.195066,11.466442,71.736411,28.263589,19.935152,8.328437,both
3,2022,Urbana,291151.0,214045.0,76983.0,123.0,73.516835,26.440919,60.899214,53.510157,12.133310,78.945941,21.054059,15.412405,5.641654,both


In [163]:
datos_finales["_merge"].value_counts()
# reviso que todos los datos hayan coincidido en las tres fuentes

,count
_merge,
both,4
left_only,0
right_only,0


In [164]:
datos_finales = datos_finales.drop(columns=["_merge"])
# elimino la columna del indicador luego de comprobar que la union fue correcta

datos_finales

,Año,Zona,MFT,APT,AZT,RET,PORCENTAJE_APROBADOS,PORCENTAJE_APLAZADOS,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo,No_pobres,Pobres,Pobreza_no_extrema,Pobreza_extrema
0,2021,Rural,98618.0,74460.0,24158.0,0.0,75.503458,24.496542,56.595429,48.467834,14.367144,73.651675,26.348325,18.007161,8.341164
1,2021,Urbana,301856.0,224383.0,77444.0,29.0,74.334451,25.655942,61.436052,51.380254,16.359620,78.233883,21.766117,16.266310,5.499806
2,2022,Rural,96624.0,72595.0,24029.0,0.0,75.131437,24.868563,56.698778,50.195066,11.466442,71.736411,28.263589,19.935152,8.328437
3,2022,Urbana,291151.0,214045.0,76983.0,123.0,73.516835,26.440919,60.899214,53.510157,12.133310,78.945941,21.054059,15.412405,5.641654


In [165]:
datos_finales["Fecha"] = pd.to_datetime(
    datos_finales["Año"].astype(str),
    format="%Y",
    errors="coerce"
)

# convierto la columna Año de numero a texto para poder utilizar el formato %Y
# y luego la convierto a datetime
# como los datos originales solo contienen el año, pandas utiliza el 1 de enero
# como fecha de referencia, por lo que no representa la fecha exacta de los datos
# si algun valor no se puede convertir, errors="coerce" lo transforma en NaT

datos_finales

,Año,Zona,MFT,APT,AZT,RET,PORCENTAJE_APROBADOS,PORCENTAJE_APLAZADOS,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo,No_pobres,Pobres,Pobreza_no_extrema,Pobreza_extrema,Fecha
0,2021,Rural,98618.0,74460.0,24158.0,0.0,75.503458,24.496542,56.595429,48.467834,14.367144,73.651675,26.348325,18.007161,8.341164,2021-01-01
1,2021,Urbana,301856.0,224383.0,77444.0,29.0,74.334451,25.655942,61.436052,51.380254,16.359620,78.233883,21.766117,16.266310,5.499806,2021-01-01
2,2022,Rural,96624.0,72595.0,24029.0,0.0,75.131437,24.868563,56.698778,50.195066,11.466442,71.736411,28.263589,19.935152,8.328437,2022-01-01
3,2022,Urbana,291151.0,214045.0,76983.0,123.0,73.516835,26.440919,60.899214,53.510157,12.133310,78.945941,21.054059,15.412405,5.641654,2022-01-01


In [166]:
datos_finales = datos_finales.rename(
    columns={
        "MFT": "Matricula_final",
        "APT": "Aprobados",
        "AZT": "Aplazados",
        "RET": "Retirados",
        "PORCENTAJE_APROBADOS": "Porcentaje_aprobados",
        "PORCENTAJE_APLAZADOS": "Porcentaje_aplazados"
    }
)

# cambio los nombres de las variables del MEP para mantener el mismo formato
# en las columnas de la base de datos final

datos_finales

,Año,Zona,Matricula_final,Aprobados,Aplazados,Retirados,Porcentaje_aprobados,Porcentaje_aplazados,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo,No_pobres,Pobres,Pobreza_no_extrema,Pobreza_extrema,Fecha
0,2021,Rural,98618.0,74460.0,24158.0,0.0,75.503458,24.496542,56.595429,48.467834,14.367144,73.651675,26.348325,18.007161,8.341164,2021-01-01
1,2021,Urbana,301856.0,224383.0,77444.0,29.0,74.334451,25.655942,61.436052,51.380254,16.359620,78.233883,21.766117,16.266310,5.499806,2021-01-01
2,2022,Rural,96624.0,72595.0,24029.0,0.0,75.131437,24.868563,56.698778,50.195066,11.466442,71.736411,28.263589,19.935152,8.328437,2022-01-01
3,2022,Urbana,291151.0,214045.0,76983.0,123.0,73.516835,26.440919,60.899214,53.510157,12.133310,78.945941,21.054059,15.412405,5.641654,2022-01-01


In [184]:
aprobados_arreglo = np.array(datos_finales["Porcentaje_aprobados"])

aprobados_arreglo
# convierto los porcentajes de aprobados en un arreglo de numpy

array([75.50345779, 74.33445086, 75.13143732, 73.51683491])

In [185]:
aprobados_arreglo[0]
# selecciono el primer porcentaje de aprobados del arreglo

np.float64(75.50345778661097)

In [186]:
aprobados_mayores_75 = aprobados_arreglo[aprobados_arreglo > 75]

aprobados_mayores_75
# filtro los porcentajes de aprobados que son mayores a 75

array([75.50345779, 75.13143732])

In [188]:
indicadores_arreglo = np.array(
    datos_finales[["Porcentaje_aprobados", "Pobres"]]
)

indicadores_arreglo
# creo un arreglo de dos dimensiones con los porcentajes de aprobacion y pobreza

array([[75.50345779, 26.34832536],
       [74.33445086, 21.76611691],
       [75.13143732, 28.26358867],
       [73.51683491, 21.05405876]])

In [190]:
aprobados_stack = aprobados_zona[["Rural", "Urbana"]].stack().rename(
    "Porcentaje_aprobados"
)

aprobados_stack
# reorganizo los porcentajes de aprobados utilizando stack
# y cambio el nombre de los valores para identificarlos mejor

Año   Zona  
2021  Rural     75.503458
      Urbana    74.334451
2022  Rural     75.131437
      Urbana    73.516835
Name: Porcentaje_aprobados, dtype: float64

## Análisis de los datos integrados

A partir de la integración de las tres fuentes se comparan los indicadores educativos y socioeconómicos entre las zonas urbana y rural durante 2021 y 2022.

### Rendimiento académico entre las zonas urbana y rural

Para comparar el rendimiento académico se utilizan los porcentajes de estudiantes aprobados y aplazados de cada zona durante 2021 y 2022.

In [167]:
comparacion_rendimiento = datos_finales[
    ["Año", "Zona", "Porcentaje_aprobados", "Porcentaje_aplazados"]
]

comparacion_rendimiento
# selecciono las variables necesarias para comparar el rendimiento entre zonas

,Año,Zona,Porcentaje_aprobados,Porcentaje_aplazados
0,2021,Rural,75.503458,24.496542
1,2021,Urbana,74.334451,25.655942
2,2022,Rural,75.131437,24.868563
3,2022,Urbana,73.516835,26.440919


In [168]:
aprobados_zona = comparacion_rendimiento.set_index(
    ["Año", "Zona"]
)["Porcentaje_aprobados"].unstack()

aprobados_zona
# organizo los porcentajes de aprobados para comparar ambas zonas por año

Zona,Rural,Urbana
Año,,
2021,75.503458,74.334451
2022,75.131437,73.516835


In [169]:
aprobados_zona["Diferencia"] = (
    aprobados_zona["Rural"] - aprobados_zona["Urbana"]
)

aprobados_zona
# calculo la diferencia en puntos porcentuales entre la zona rural y urbana

Zona,Rural,Urbana,Diferencia
Año,,,
2021,75.503458,74.334451,1.169007
2022,75.131437,73.516835,1.614602


In [170]:
aplazados_zona = comparacion_rendimiento.set_index(
    ["Año", "Zona"]
)["Porcentaje_aplazados"].unstack()

aplazados_zona
# organizo los porcentajes de aplazados para comparar ambas zonas por año

Zona,Rural,Urbana
Año,,
2021,24.496542,25.655942
2022,24.868563,26.440919


In [171]:
aplazados_zona["Diferencia"] = (
    aplazados_zona["Urbana"] - aplazados_zona["Rural"]
)

aplazados_zona
# calculo la diferencia en puntos porcentuales de aplazados entre la zona urbana y rural

Zona,Rural,Urbana,Diferencia
Año,,,
2021,24.496542,25.655942,1.159400
2022,24.868563,26.440919,1.572356


### Interpretación del rendimiento académico

Durante 2021 y 2022 la zona rural presentó un porcentaje de aprobación mayor que la zona urbana. En 2021 la diferencia fue de aproximadamente 1.17% y en 2022 aumentó a 1.61%.

Por otra parte, la zona urbana presentó un mayor porcentaje de estudiantes aplazados. La diferencia fue de aproximadamente 1.16% en 2021 y 1.57% en 2022.

Por lo tanto, durante los dos años analizados se observa un rendimiento académico ligeramente más positivo en la zona rural según los porcentajes de aprobación y aplazamiento.

### Indicadores de empleo y pobreza entre las zonas urbana y rural

Para comparar las condiciones socioeconómicas se utilizan las tasas de participación, ocupación y desempleo, junto con el porcentaje de pobreza de cada zona durante 2021 y 2022.

In [172]:
comparacion_socioeconomica = datos_finales[
    ["Año", "Zona", "Tasa_participacion", "Tasa_ocupacion",
     "Tasa_desempleo", "Pobres"]
]

comparacion_socioeconomica
# selecciono los indicadores de empleo y pobreza que voy a comparar entre zonas

,Año,Zona,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo,Pobres
0,2021,Rural,56.595429,48.467834,14.367144,26.348325
1,2021,Urbana,61.436052,51.380254,16.359620,21.766117
2,2022,Rural,56.698778,50.195066,11.466442,28.263589
3,2022,Urbana,60.899214,53.510157,12.133310,21.054059


In [173]:
participacion_zona = comparacion_socioeconomica.set_index(
    ["Año", "Zona"]
)["Tasa_participacion"].unstack()

participacion_zona
# organizo las tasas de participacion para comparar ambas zonas por año

Zona,Rural,Urbana
Año,,
2021,56.595429,61.436052
2022,56.698778,60.899214


In [174]:
participacion_zona["Diferencia"] = (
    participacion_zona["Urbana"] - participacion_zona["Rural"]
)

participacion_zona
# calculo la diferencia en puntos porcentuales de participacion entre la zona urbana y rural

Zona,Rural,Urbana,Diferencia
Año,,,
2021,56.595429,61.436052,4.840623
2022,56.698778,60.899214,4.200436


In [175]:
ocupacion_zona = comparacion_socioeconomica.set_index(
    ["Año", "Zona"]
)["Tasa_ocupacion"].unstack()

ocupacion_zona
# organizo las tasas de ocupacion para comparar ambas zonas por año

Zona,Rural,Urbana
Año,,
2021,48.467834,51.380254
2022,50.195066,53.510157


In [176]:
ocupacion_zona["Diferencia"] = (
    ocupacion_zona["Urbana"] - ocupacion_zona["Rural"]
)

ocupacion_zona
# calculo la diferencia en puntos porcentuales de ocupacion entre la zona urbana y rural

Zona,Rural,Urbana,Diferencia
Año,,,
2021,48.467834,51.380254,2.912420
2022,50.195066,53.510157,3.315091


In [177]:
desempleo_zona = comparacion_socioeconomica.set_index(
    ["Año", "Zona"]
)["Tasa_desempleo"].unstack()

desempleo_zona
# organizo las tasas de desempleo para comparar ambas zonas por año

Zona,Rural,Urbana
Año,,
2021,14.367144,16.35962
2022,11.466442,12.13331


In [178]:
desempleo_zona["Diferencia"] = (
    desempleo_zona["Urbana"] - desempleo_zona["Rural"]
)

desempleo_zona
# calculo la diferencia en puntos porcentuales de desempleo entre la zona urbana y rural

Zona,Rural,Urbana,Diferencia
Año,,,
2021,14.367144,16.35962,1.992476
2022,11.466442,12.13331,0.666868


In [179]:
pobreza_zona = comparacion_socioeconomica.set_index(
    ["Año", "Zona"]
)["Pobres"].unstack()

pobreza_zona
# organizo los porcentajes de pobreza para comparar ambas zonas por año

Zona,Rural,Urbana
Año,,
2021,26.348325,21.766117
2022,28.263589,21.054059


In [180]:
pobreza_zona["Diferencia"] = (
    pobreza_zona["Rural"] - pobreza_zona["Urbana"]
)

pobreza_zona
# calculo la diferencia en puntos porcentuales de pobreza entre la zona rural y urbana

Zona,Rural,Urbana,Diferencia
Año,,,
2021,26.348325,21.766117,4.582208
2022,28.263589,21.054059,7.209530


### Interpretación de los indicadores de empleo y pobreza

Durante 2021 y 2022 la zona urbana presentó mayores tasas de participación y ocupación que la zona rural. La diferencia en participación fue de aproximadamente 4.84% en 2021 y 4.20% en 2022, mientras que la diferencia en ocupación pasó de 2.91% a 3.32%.

En cuanto al desempleo, la zona urbana presentó una tasa mayor en ambos años. Sin embargo, la diferencia entre las zonas disminuyó de aproximadamente 1.99% en 2021 a 0.67% en 2022. Además, la tasa de desempleo disminuyó en ambas zonas durante este periodo.

Por otra parte, la pobreza fue mayor en la zona rural. La diferencia con respecto a la zona urbana fue de aproximadamente 4.58% en 2021 y aumentó a 7.21% en 2022. Mientras la pobreza rural aumentó entre ambos años, la pobreza urbana disminuyó ligeramente.

En general, los datos muestran diferencias socioeconómicas entre ambas zonas: la zona urbana presentó mayores niveles de participación y ocupación, mientras que la zona rural presentó mayores niveles de pobreza.

### Relación entre el rendimiento académico y las condiciones socioeconómicas

Para observar la relación entre el rendimiento académico y las condiciones socioeconómicas se comparan los porcentajes de aprobación con los indicadores de empleo y pobreza de cada zona durante 2021 y 2022.

In [181]:
relacion_indicadores = datos_finales[
    ["Año", "Zona", "Porcentaje_aprobados",
     "Tasa_participacion", "Tasa_ocupacion",
     "Tasa_desempleo", "Pobres"]
]

relacion_indicadores
# selecciono los indicadores educativos y socioeconomicos para observar su relacion

,Año,Zona,Porcentaje_aprobados,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo,Pobres
0,2021,Rural,75.503458,56.595429,48.467834,14.367144,26.348325
1,2021,Urbana,74.334451,61.436052,51.380254,16.359620,21.766117
2,2022,Rural,75.131437,56.698778,50.195066,11.466442,28.263589
3,2022,Urbana,73.516835,60.899214,53.510157,12.133310,21.054059


In [182]:
aprobacion_pobreza = relacion_indicadores[
    ["Año", "Zona", "Porcentaje_aprobados", "Pobres"]
]

aprobacion_pobreza
# comparo el porcentaje de aprobados con el porcentaje de pobreza de cada zona

,Año,Zona,Porcentaje_aprobados,Pobres
0,2021,Rural,75.503458,26.348325
1,2021,Urbana,74.334451,21.766117
2,2022,Rural,75.131437,28.263589
3,2022,Urbana,73.516835,21.054059


In [183]:
aprobacion_empleo = relacion_indicadores[
    ["Año", "Zona", "Porcentaje_aprobados",
     "Tasa_participacion", "Tasa_ocupacion", "Tasa_desempleo"]
]

aprobacion_empleo
# comparo el porcentaje de aprobados con los indicadores de empleo de cada zona

,Año,Zona,Porcentaje_aprobados,Tasa_participacion,Tasa_ocupacion,Tasa_desempleo
0,2021,Rural,75.503458,56.595429,48.467834,14.367144
1,2021,Urbana,74.334451,61.436052,51.380254,16.359620
2,2022,Rural,75.131437,56.698778,50.195066,11.466442
3,2022,Urbana,73.516835,60.899214,53.510157,12.133310


### Interpretación de la relación entre rendimiento y condiciones socioeconómicas

Al comparar los indicadores de las tres fuentes, no se observa una relación directa entre mejores condiciones socioeconómicas y un mayor porcentaje de aprobación.

Durante 2021 y 2022 la zona urbana presentó mayores tasas de participación y ocupación y menores niveles de pobreza. Sin embargo, la zona rural presentó porcentajes de aprobación ligeramente mayores en ambos años.

En 2022, por ejemplo, la pobreza rural aumentó hasta aproximadamente 28.26%, mientras que el porcentaje de aprobación se mantuvo en 75.13%. En la zona urbana la pobreza fue menor, con aproximadamente 21.05%, pero el porcentaje de aprobación fue de 73.52%.

Por lo tanto, para las zonas urbana y rural durante los años analizados, las diferencias observadas en los indicadores de empleo y pobreza no muestran una relación directa con las diferencias en el porcentaje de aprobación. Estos resultados permiten identificar patrones entre las zonas, pero no establecer que las condiciones socioeconómicas sean la causa de las diferencias en el rendimiento académico.

In [191]:
datos_finales.to_csv(
    "/content/datos_finales.csv",
    index=False
)

# exporto la base de datos final en formato csv

In [192]:
datos_finales.to_json(
    "/content/datos_finales.json",
    orient="records",
    force_ascii=False
)

# exporto la base de datos final en formato json